common/tls/src/main/java/io/helidon/common/tls/ConfiguredTlsManager.java

```diff
@@ -396,61 +245,143 @@ static KeyStore keystore(TlsConfig.BuilderBase<?, ?> target) {
                  | NoSuchAlgorithmException
                  | CertificateException e) {
             throw new IllegalArgumentException("Invalid configuration of internal keystores. Provider: "
-                                                       + target.internalKeystoreProvider()
-                                                       + ", type: " + target.internalKeystoreType(), e);
+                                                       + tlsConfig.internalKeystoreProvider()
+                                                       + ", type: " + tlsConfig.internalKeystoreType(), e);
         }
     }
 
-    static TrustManagerFactory tmf(TlsConfig.BuilderBase<?, ?> target) {
+    /**
+     * Create a new trust manager factory based on the configuration.
+     *
+     * @param tlsConfig TLS Config
+     * @return a new trust manager factory
+     */
+    protected TrustManagerFactory tmf(TlsConfig tlsConfig) {
         try {
-            String algorithm = target.trustManagerFactoryAlgorithm().orElseGet(TrustManagerFactory::getDefaultAlgorithm);
-            if (target.trustManagerFactoryProvider().isEmpty()) {
+            String algorithm = tlsConfig.trustManagerFactoryAlgorithm().orElseGet(TrustManagerFactory::getDefaultAlgorithm);
+            if (tlsConfig.trustManagerFactoryProvider().isEmpty()) {
                 return TrustManagerFactory.getInstance(algorithm);
             } else {
-                return TrustManagerFactory.getInstance(algorithm, target.trustManagerFactoryProvider().get());
+                return TrustManagerFactory.getInstance(algorithm, tlsConfig.trustManagerFactoryProvider().get());
             }
         } catch (NoSuchAlgorithmException | NoSuchProviderException e) {
             throw new IllegalArgumentException("Invalid configuration of trust manager factory. Provider: "
-                                                       + target.trustManagerFactoryProvider()
-                                                       + ", algorithm: " + target.trustManagerFactoryAlgorithm(), e);
+                                                       + tlsConfig.trustManagerFactoryProvider()
+                                                       + ", algorithm: " + tlsConfig.trustManagerFactoryAlgorithm(), e);
         }
     }
 
-    static ReloadableData<KeyManager, X509KeyManager> wrapX509KeyManagers(List<TlsReloadableComponent> reloadable,
-                                                                          KeyManager[] keyManagers) {
+    private void sslContext(TlsConfig tlsConfig) {
```

Comment:
perhaps a better name would be to overload `initSslContext`

Reply:
Not really, as this is a method that is specific to the configured manager (and it does call the `initSslContext` as the last thing). I just did not want to pollute the `init(TlsConfig)` method with the code that is in this method.